docs/parameters.yaml

```diff
@@ -2196,6 +2196,13 @@ type: bool
 default: false
 components: ["cache", "director", "origin", "registry"]
 ---
+name: Server.DbLocation
+description: |+
+  A filepath to the intended location of the server's database.
+type: filename
+default: /var/lib/pelican/server.sqlite
+components: ["origin", "cache", "registry", "director"]
+---
```

Comment:
I think I was okay with this approach until I realized the Origin now has two sqlite databases. Previously we talked about using a common database package for Pelican, and we even talked about combining all our sqlite into a single database. I think that's outside the scope of this pull request, but at the very least let's (for now) make a separate `Cache.DbLocation` instead of introducing the separate `Server.DbLocation`.

When you introduce this, check the other `*.DbLocation` defaults, as we should set the default location to conform to that pattern. The actual instantiation of the default should happen in `config/config.go` and not in `config/resources/defaults.yaml`, which doesn't let you do dynamic defaulting.